### ISSUE TITLE
fig.suptitle gives faulty output

### ISSUE THREAD

**Issue by snim2** [nschloe/matplotlib2tikz#64]
The `.tikz` file here: https://gist.github.com/snim2/eec246862c72fdca4391#file-example-tikz was generated with `matplotlib2tikz` on a Ubuntu 15.04 machine, with LaTeX packages as per the file list below.

Running `pdflatex` on this LaTeX file based on the example in the manual:

```latex
\documentclass[a4paper]{article}
\usepackage{tikz}
\usetikzlibrary{positioning,calc}
\usepackage{pgf}
\usepackage{pgfplots}
\usepgfplotslibrary{groupplots}
\listfiles
\newlength\figureheight{}
\newlength\figurewidth{}
\setlength\figureheight{4cm}
\setlength\figurewidth{6cm}

\begin{document}
My pgf version is: \pgfversion
\input{example.tikz}
\end{document}
```

the document fails to compile with this error:

```latex
! Missing \endcsname inserted.
<to be read again> 
                   \pgfcrdmth@x 
l.8106 \node at (axis cs:0.5,0.98)
```

Commenting out lines 8106--8111 of the `.tikz` file results in the PDF here: https://drive.google.com/file/d/0B9T1QPEACWUlSTlJSmRtT0VRYjQ/view?usp=sharing

It looks to me as if `matplotlib2tikz` should be generating a bit more code to define an axis, or have I missed a pgf library somewhere?

**File list**

```tex
 *File List*
 article.cls    2014/09/29 v1.4h Standard LaTeX document class
  size10.clo    2014/09/29 v1.4h Standard LaTeX file (size option)
    tikz.sty    2013/12/13 v3.0.0 (rcs-revision 1.142)
     pgf.sty    2013/12/18 v3.0.0 (rcs-revision 1.14)
  pgfrcs.sty    2013/12/20 v3.0.0 (rcs-revision 1.28)
everyshi.sty    2001/05/15 v3.00 EveryShipout Package (MS)
  pgfrcs.code.tex
 pgfcore.sty    2010/04/11 v3.0.0 (rcs-revision 1.7)
graphicx.sty    2014/04/25 v1.0g Enhanced LaTeX Graphics (DPC,SPQR)
  keyval.sty    2014/05/08 v1.15 key=value parser (DPC)
graphics.sty    2009/02/05 v1.0o Standard LaTeX Graphics (DPC,SPQR)
    trig.sty    1999/03/16 v1.09 sin cos tan (DPC)
graphics.cfg    2010/04/23 v1.9 graphics configuration of TeX Live
  pdftex.def    2011/05/27 v0.06d Graphics/color for pdfTeX
infwarerr.sty    2010/04/08 v1.3 Providing info/warning/error messages (HO)
 ltxcmds.sty    2011/11/09 v1.22 LaTeX kernel commands for general use (HO)
  pgfsys.sty    2013/11/30 v3.0.0 (rcs-revision 1.47)
  pgfsys.code.tex
pgfsyssoftpath.code.tex    2013/09/09  (rcs-revision 1.9)
pgfsysprotocol.code.tex    2006/10/16  (rcs-revision 1.4)
  xcolor.sty    2007/01/21 v2.11 LaTeX color extensions (UK)
   color.cfg    2007/01/18 v1.5 color configuration of teTeX/TeXLive
 pgfcore.code.tex
pgfcomp-version-0-65.sty    2007/07/03 v3.0.0 (rcs-revision 1.7)
pgfcomp-version-1-18.sty    2007/07/23 v3.0.0 (rcs-revision 1.1)
  pgffor.sty    2013/12/13 v3.0.0 (rcs-revision 1.25)
 pgfkeys.sty    
 pgfkeys.code.tex
 pgfmath.sty    
 pgfmath.code.tex
  pgffor.code.tex
    tikz.code.tex
tkz-euclide.sty    2011/06/01 1.16 c for plots 2d 
tkz-base.sty    2011/06/01 1.16 c tkz-base
    etex.sty    1998/03/26 v2.0 eTeX basic definition package (PEB)
numprint.sty    2012/08/20 v1.39 Print numbers (HH)
   array.sty    2014/10/28 v2.4c Tabular extension package (FMi)
      fp.sty    1995/04/02
defpattern.sty    1994/10/12
fp-basic.sty    1996/05/13
fp-addons.sty    1995/03/15
 fp-snap.sty    1995/04/05
  fp-exp.sty    1995/04/03
fp-trigo.sty    1995/04/14
  fp-pas.sty    1994/08/29
fp-random.sty    1995/02/23
  fp-eqn.sty    1995/04/03
  fp-upn.sty    1996/10/21
 fp-eval.sty    1995/04/03
tkz-tools-utilities.tex
tkz-tools-arith.tex
tkz-tools-base.tex
tkz-tools-misc.tex
tkz-tools-math.tex
tkz-obj-points.tex
tkz-obj-segments.tex
tkz-obj-marks.tex
tkz-tools-intersections.tex
tkz-tools-transformations.tex
tkz-obj-lines.tex
tkz-lib-symbols.tex
tkz-obj-circles.tex
tkz-obj-addpoints.tex
pgfplots.sty    2014/08/04 v1.11 Data Visualization (1.11-3-g4c99d7a)
supp-pdf.mkii
pdftexcmds.sty    2011/11/29 v0.20 Utility functions of pdfTeX for LuaTeX (HO)
ifluatex.sty    2010/03/01 v1.3 Provides the ifluatex switch (HO)
   ifpdf.sty    2011/01/30 v2.3 Provides the ifpdf switch (HO)
epstopdf-base.sty    2010/02/09 v2.5 Base part for package epstopdf
  grfext.sty    2010/08/19 v1.1 Manage graphics extensions (HO)
kvdefinekeys.sty    2011/04/07 v1.3 Define keys (HO)
kvoptions.sty    2011/06/30 v3.11 Key value format for package options (HO)
kvsetkeys.sty    2012/04/25 v1.16 Key value parser (HO)
etexcmds.sty    2011/02/16 v1.5 Avoid name clashes with e-TeX commands (HO)
epstopdf-sys.cfg    2010/07/13 v1.3 Configuration of (r)epstopdf for TeX Live
```

/cc @vext01

**Comment by nschloe** [comment]
Thanks for the report! Could you provide a little Python script that generates the faulty TikZ file when run through matplotlib2tikz? We could add that to our test suite to make sure the error won't crop up anymore once we've fixed it.
Also: What pgfplots version are you using? (It usually shows in the TeX compile output.)

**Comment by snim2** [comment]
Thanks. The PDF file shows version 3.0.0 (from `\pgfversion`); watch this space for a Python script...

**Comment by nschloe** [comment]
This is the PGF (aka TikZ) version. PGFPlots is a library that sits on top of it.

**Comment by snim2** [comment]
Apologies, `pgfplots.revision.tex` says `1.11`

**Comment by nschloe** [comment]
`1.12.1 (2015/05/02)` is the most recent version, but I suspect this is a bug in matplotlib2tikz. I'll be happy to look at a python script that produces the plot. The simpler, the better. :)

**Comment by snim2** [comment]
OK, I've cut this back as far as I think it can go, and I still get the same error from LaTeX:

```python
#!/usr/bin/env python2.7

import matplotlib.pyplot as plt
from matplotlib2tikz import save as tikz_save

def draw_subplot(axis, data, title, x_range, y_range):
    axis.plot(data)
    axis.set_ylim(y_range)


def draw_plot(data, x_bounds):
    y_min, y_max = float("inf"), float("-inf")
    for dataset in data:
        y_min = min(min(dataset), y_min)
        y_max = max(max(dataset), y_max)

    fig, axes = plt.subplots(2, 2, squeeze=False)

    row = 0
    col = 0
    for _ in [0, 1]:
        for _ in [0, 1]:
            dataset = data[(row * 2) + col]
            title = "PLOT TITLE"
            axis = axes[row, col]
            axis.set_xlim(x_bounds)
            draw_subplot(axis, dataset, title, x_bounds, [y_min, y_max])
            col += 1
        row += 1
        col = 0

    fig.suptitle("PLOT TITLE", fontsize=18, fontweight="bold")
    filename_tikz = "example.tikz"
    tikz_save(filename_tikz,
              figureheight = '\\figureheight',
              figurewidth = '\\figurewidth')
    plt.close()


if __name__ == "__main__":
    import random
    plt.close()  # avoid extra blank window
    NUM = 2000
    data = [ [random.random() for _ in range(NUM)] for _ in range(4) ]
    draw_plot(data, (0, NUM))
```

**Comment by nschloe** [comment]
Thanks!
I could squeeze it down even more. This
```python
#!/usr/bin/env python

import matplotlib.pyplot as plt
from matplotlib2tikz import save as tikz_save


def draw_plot():

    fig = plt.figure()
    fig.suptitle('PLOT TITLE', fontsize=18, fontweight='bold')
    plt.plot([0.0, 1.0], [0.0, 1.0])

    # plt.show()
    tikz_save('figure.tex', figureheight='4cm', figurewidth='4cm')


if __name__ == '__main__':
    draw_plot()
```
Will already yield the error. The culprit is the `suptitle` which matplotlib2tik can't cope with at the moment.

**Comment by nschloe** [comment]
The faulty output has new been fixed, LaTeX compiles fine. The text still doesn't quite appear where it does in the mpl figure; take, e.g.,
![sp](https://cloud.githubusercontent.com/assets/181628/10718016/d4c618ae-7b68-11e5-98de-116c887f9fcb.png)
but there isn't much more on the matplotlib2tikz side that we can do here. If the super title coordinates are given as `(0.5, 0.98)`, they refer to the entire picture, and since the dimensions are slightly different in mpl and in LaTeX, a little tweaking will always be required. It should be easy to handle though.

To be included in the next release (0.2.5).
